### DISCUSSION TITLE
Progress Updates

### DISCUSSION THREAD

**Discussion by WillCodeForCats** [WillCodeForCats/tekmar-482#1] [General]
I just wanted to update everyone that I'm still here and working on the integration. I had some other life stuff going on earlier this year which took time away from working on this, but I've not abandoned it.

I initially posted updates at https://community.home-assistant.io/t/tekmar-heating-controls-fujitsu-split-a-c/68020 but that forums blocks me from posting more than 3 times in a row unless someone else responds, which is completely incompatible with posting progress updates where responses are not expected. So I'll be posting them here instead.

Most recently I came across a problem in testing where sometimes the gateway doesn't respond to requests. Although it's not a showstopper issue it's bad for the user experience because if you tell HA to do something you should expect an update from the gateway, so I want to address this before a release.

Please visit the wiki section. Documentation is being developed "on the fly" as I work on things. Once I have something that's as complete as I feel I can make it the first version of the integration will be released and we'll see what happens with bug reports. I don't have hands on discontinued thermostats and snow melt controls (as much as I'd love a heated driveway to avoid shoveling snow, I don't have one) unless someone wants to send some to me to test, so these will probably rely on bug reports if I haven't interpreted the manuals correctly.

Wiki: [https://github.com/WillCodeForCats/tekmar-482/wiki](https://github.com/WillCodeForCats/tekmar-482/wiki)

**Comment by rhl-** [comment]
Hi, I have tekmar in my house. I am thinking of purchasing a tekmar 482/486 and building some automation around my cooling system.

It looks like your working on getting the 482  working with home assistant. That seems like it could be an improvement over the 486, as the 482 + home assistant is probably also the same as the 486 UI, but then other software could be written. 

I'm curious if your project is close to release? I could help develop once i have setup my 482 + house control..

**Comment by WillCodeForCats** [comment]
I do have a fully functional integration running, but it still has a few problems. The big ones are handling messages it asked for but never got a response to, and if communication to the companion add-on is lost the integration will not recover (I want it to self-recover). There's also some minor stuff with testing the Setpoint 161 but that can be dealt with as bugfixes.

I bought both the 482 and 486 and unfortunately the 486 completely useless for home automation. It's a nice unit with a touchscreen and onboard SD card for logging data, but there's no API and it's tied to Tekmar's cloud service (no local access, all ports closed). So for the now the 482 is where it's at unless some future update to the 486 adds local access.

I should update them but here's some screenshots: https://github.com/WillCodeForCats/tekmar-482/wiki/Features

**Reply by rhl-** [reply]
Is all of your code public at this point? I've not worked at all with HA but looking at the pre-release branch i dont see much code..

Happy to help out once there is something out there.

**Comment by WillCodeForCats** [comment]
No, I haven't uploaded anything yet until it's a bit more ready. I don't like releasing things with obvious defects, plus this is my first time developing integrations and add-ons from scratch for Home Assistant so I don't want to have too many stupid mistakes in the first release.

It definitely can't work outside of Home Assistant (this is all just a plugin/module for it), so if you haven't looked at HA yet: https://www.home-assistant.io

**Reply by rhl-** [reply]
Hi, yes, i have one installed now. I wish to integrate my tekmar 482. I can help with the software side of things. There is no reason to not push to a remote before its finished. you could push and others can help as needed. just make things clear in your readme.

**Comment by safadig** [comment]
Is it python? I can’t seem to locate the developer/api from tekmar/watts.
Do you have the url? Thanks

**Reply by rhl-** [reply]
The author has code for reading the RS-232 packet data and interfacing with home assistant. He has not yet released it. the "API" from tekmar is the spec for the packets on the tekmarnet gateway.

**Comment by WillCodeForCats** [comment]
Integration v0.9.0 and [add-on v1.0.2](https://github.com/WillCodeForCats/tekmar-packetserv) were released on May 8th. There are a few known issues but outside of those it does work and has been stable for me.
